File: src/main/java/com/baulsupp/oksocial/services/microsoft/MicrosoftUtil.java
package com.baulsupp.oksocial.services.microsoft;

import com.google.common.collect.Sets;
import java.util.Collections;
import java.util.Set;
import okhttp3.Request;

public class MicrosoftUtil {
  private MicrosoftUtil() {
  }

  public static final Set<String> API_HOSTS =
      Collections.unmodifiableSet(Sets.newHashSet(
          "graph.microsoft.com")
      );

  public static Request apiRequest(String s, Request.Builder requestBuilder) {
    return requestBuilder.url("https://graph.microsoft.com" + s).build();
  }
}

File: src/main/kotlin/com/baulsupp/oksocial/services/microsoft/MicrosoftUtil.kt
package com.baulsupp.oksocial.services.microsoft

import okhttp3.Request

object MicrosoftUtil {

    val API_HOSTS = setOf((
            "graph.microsoft.com")
    )

    fun apiRequest(s: String, requestBuilder: Request.Builder): Request {
        return requestBuilder.url("https://graph.microsoft.com" + s).build()
    }
}
